Extending/Packages/Packages-on-Umbraco-Cloud/index.md

```diff
@@ -4,9 +4,42 @@ versionFrom: 8.0.0
+If you need a db table in each environments db you will probably need to look at migrations below instead.
+
+### Migrations
+
+A [database migration](../../Database/index.md) is a set of database commands you run as part of a migration plan. That migration plan has an id that is stored in the database (in the KeyValue table). This means that when you add new migrations Umbraco will only execute the ones that came after the one with the stored id.
+
+As database migrations are run in the database of the site it also means that they will run on each environment you trigger them on. The most common way to trigger a db migration is to include them in a [composer](../../../Implementation/Composing/index.md), which will ensure they run on site startup. This means any database commands you have in your migration will automatically run when the site starts up. When your package code is pushed to a new environment it will run them from the beginning on that environment as no id is saved in the database.
+
+This is normally a good thing, however if you generate any Umbraco schema then Deploy will automatically create [UDA files](../../../Umbraco-Cloud/Set-Up/Power-Tools/Generating-UDA-files/index.md#what-are-uda-files) based on that schema, and commit them to source control. Which means if you then deploy all your files to the next environment the migration will run again, create duplicates and generate duplicate UDA files - will end up causing a lot of issues.
+
+You could consider creating Umbraco schema only during a package action, and then running things like creating db tables in migrations. Another good workaround could be to not run the migrations in a composer, but rather create a dashboard for the package where the user can choose which migrations to run themselves. The package [Articulate has an example of this](https://github.com/Shazwazza/Articulate/blob/master/build/packageManifest.xml#L613).
+
+### Creating files
+
+You may sometimes choose to save data in a file, could be a seperate config file for your package or a [config transform file](../../..//Umbraco-Cloud/Set-Up/Config-Transforms/index.md) to fx. add an app setting to web.config.
+If you do this be aware of two things:
+1.  If these files are generated on a Cloud environment they will not be stored in source control, and will be overwritten on next deployment. They need to be installed locally, committed to source control and then pushed up to the Cloud environments. We have an [existing feature request](https://github.com/umbraco/Umbraco.Cloud.Issues/issues/33) on allowing package creators to commit their files directly on Cloud, and it is possible to do so currently but not in a supported way, and it may change suddenly.
```

Comment:
I like that you mention the existing feature request here, as it makes a lot of sense..
But I do have one concern: How are you keeping track of this? Like, how do we ensure that this articles is updated, once the feature request is realized / added? 😅

Reply:
Can't guarantee that we will, up to you if it's not worth it 😁

Reply:
Let's just put it in for now 💪